v1/retryable_client.go

```diff
@@ -0,0 +1,240 @@
+package authzed
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"github.com/cenkalti/backoff/v4"
+	"github.com/samber/lo"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"strings"
+	"time"
+)
+
+// ConflictStrategy is an enumeration type that represents the strategy to be used
+// when a conflict occurs during a bulk import of relationships in Authzed.
+// The strategies are as follows:
+// - Fail: The operation will fail if any duplicate relationships are found.
+// - Skip: The operation will ignore duplicates and continue with the import.
+// - Touch: The operation will retry the import with TOUCH semantics in case of duplicates.
+type ConflictStrategy int
+
+const (
+	Fail ConflictStrategy = iota
+	Skip
+	Touch
+
+	defaultBackoff    = 50 * time.Millisecond
+	defaultMaxRetries = 10
+)
+
+// Fallback for datastore implementations on SpiceDB < 1.29.0 not returning proper gRPC codes
+// Remove once https://github.com/authzed/spicedb/pull/1688 lands
+var (
+	txConflictCodes = []string{
+		"SQLSTATE 23505",     // CockroachDB
+		"Error 1062 (23000)", // MySQL
+	}
+	retryableErrorCodes = []string{
+		"retryable error",                          // CockroachDB, PostgreSQL
+		"try restarting transaction", "Error 1205", // MySQL
+	}
+)
+
+// RetryableClient represents an open connection to Authzed with
```

Comment:
SpiceDB

Reply:
I'll fix this, but I was following the pattern set on the doc comments for the other clients. They all say Authzed.

Reply:
Yeah, we're transitioning those to make it match the fact that this is open source